components/loggingInterop.ts

```diff
@@ -19,6 +19,44 @@
 import type { NextFunction, Response } from "express"
 import type { RequestWithJwt } from "./types/types"
 import picocolors from "picocolors"
+import winston from "winston"
+import "winston-daily-rotate-file"
+
+const isDebug = ["*", "true", "peacock", "yes"].includes(
+    process.env.DEBUG || "false",
+)
+
+const fileTransport = new winston.transports.DailyRotateFile({
+    filename: "logs/application-%DATE%.json",
+    datePattern: "YYYYMMDDHHmmss",
+    frequency: "1d",
+    maxFiles: "14d",
+    format: winston.format.printf((info) => {
+        return JSON.stringify(info.data)
+    }),
+})
+
+const consoleTransport = new winston.transports.Console({
+    format: winston.format.printf((info) => {
+        if (info.stack) {
+            return `${info.message}\n${info.stack}`
+        }
+
+        return info.message
+    }),
+})
+
+const logger = winston.createLogger({
+    levels: {
+        error: 0,
+        warn: 1,
+        info: 2,
+        debug: 3,
+        trace: 4,
```

Comment:
Can we use the LogLevel enum here?

Reply:
Check the next commit - this is the best we can do with Winston.